windows/windows_online_updates_install/windows_online_updates_install.yml

```diff
@@ -0,0 +1,106 @@
+
+        - name: "Test setup"
+          include_tasks: ../setup/test_setup.yml
+
+        - name: "Get guest OS version build before installing Windows updates"
+          include_tasks: ../utils/win_get_os_version.yml
+        - name: "Display guest OS version build before installing Windows updates"
+          ansible.builtin.debug: var=win_os_version_build
+        - name: "Set fact of guest OS build number before installing Windows updates"
+          ansible.builtin.set_fact:
+            win_os_build_before_update: "{{ ('.').join(win_os_version_build.split('.')[2:]) }}"
+        - name: "Check expected OS build number"
+          ansible.builtin.assert:
+            that:
+              - win_os_build_before_update is version(windows_updated_build_num, '<=')
+            fail_msg: >-
+              Before installing Windows updates, guest OS build number is '{{ win_os_build_before_update }}',
+              which is newer than the configured expected build number '{{ windows_updated_build_num }}',
+              please check and change parameter value 'windows_updated_build_num' in vars/test.yml.
+          when: windows_updated_build_num is defined
+
+        - name: "Install Windows online updates in guest OS"
+          include_tasks: install_windows_updates.yml
+
+        - name: "No Windows updates installed in guest OS"
+          ansible.builitin.fail:
+            msg: >-
+              Before installing Windows updates, guest OS build number is '{{ win_os_build_before_update }}',
+              configured expected build number is '{{ windows_updated_build_num }}', while after trying install
+              Windows updates in guest OS, no update installed.
+          when:
+            - windows_updated_build_num is defined
+            - not win_online_updates_installed
+
+        - name: "Windows online updates installed in guest OS"
+          when: win_online_updates_installed
+          block:
+            - name: "Get guest OS version build after installing Windows updates"
+              include_tasks: ../utils/win_get_os_version.yml
+            - name: "Display guest OS version build after installing Windows updates"
+              ansible.builtin.debug: var=win_os_version_build
+            - name: "Set fact of the guest OS build number after installing Windows updates"
+              ansible.builtin.set_fact:
+                win_os_build_after_update: "{{ ('.').join(win_os_version_build.split('.')[2:]) }}"
+
+            - name: "Check OS build number after installing Windows updates"
+              ansible.builtin.assert:
+                that:
+                  - win_os_build_after_update is version(windows_updated_build_num, '>=')
+                fail_msg: >-
+                  After installing Windows updates, guest OS build number is '{{ win_os_build_after_update }}',
+                  which is not newer than the configured expected build '{{ windows_updated_build_num }}'.
```

Comment:
I suggest changing to `which is not expected build '{{ windows_updated_build_num }}' or newer`.

Reply:
Modified, thanks.